app/services/course/statistics/assessments_score_summary_download_service.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+require 'csv'
+class Course::Statistics::AssessmentsScoreSummaryDownloadService
+  include ApplicationFormattersHelper
+
+  class << self
+    def download(course, assessment_ids)
+      service = new(course, assessment_ids)
+      ActsAsTenant.without_tenant do
+        service.generate_csv_report
+      end
+    end
+  end
+
+  def generate_csv_report
+    assessment_file_name = "#{Pathname.normalize_filename(@course.title)}_assessment_score_summary.csv"
+    assessment_score_summary_file_path = File.join(@base_dir, assessment_file_name)
+
+    load_total_grades
+    CSV.open(assessment_score_summary_file_path, 'w') do |csv|
+      download_score_summary(csv)
+    end
+
+    assessment_score_summary_file_path
+  end
+
+  private
+
+  def initialize(course, assessment_ids)
+    @course = course
+    @assessment_ids = assessment_ids
+    @base_dir = Dir.mktmpdir('assessment-score-summary-')
```

Comment:
Looks ripe for name clash, can you check what happens when multiple score summary downloads are happening concurrently?

Reply:
depends on the browser, but generally if there are 2 name clashing for download, the latter one will be automatically be added `(1)` as a suffix (or 2, 3, etc. depending on how many names are clashing)